services/core/ready.js

```diff
@@ -0,0 +1,76 @@
+/*
+ * LiskHQ/lisk-service
+ * Copyright © 2021 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const logger = require('lisk-service-framework').Logger();
+
+const core = require('./shared/core');
+const signals = require('./shared/signals');
+const config = require('./config');
+
+const features = {
+    isIndexReady: false,
+    isTransactionStatsReady: false,
+    isFeeEstimatesReady: false,
+    isDelegatesReady: false,
+    // IsPeersReady: false,
+};
+
+const isCoreReady = () => !Object.keys(features).some(value => !features[value]);
+
+// Check if all blocks are indexed
+signals.get('blockIndexReady').add(() => {
+    logger.debug('Indexing finished');
+    features.isIndexReady = true;
+});
+
+// Check if transaction stats are built
+signals.get('transactionStatsReady').add((days) => {
+    logger.debug('Transaction stats calculated for:', `${days}days`);
+    features.isTransactionStatsReady = true;
+});
+
+// Check if peers are tracked in real time
+// signals.get('peerReload').add(async (peers) => {
+//     const currentHeight = (await core.getNetworkStatus()).data.height;
+//     if (peers.length) {
+//         const isPeersReady = peers[0].height === currentHeight;
+//         features.IsPeersReady = isPeersReady;
+//     }
+// });
```

Comment:
Is that code needed?

Reply:
Actually, that logic is to verify peers built are real time or not. But as we discussed before, we dont need this. I will remove  it.

Reply:
Right, it can be removed. Peer retrieval won't delay start significantly and is not needed on CI.